src/lib.rs

```diff
@@ -31,3 +31,54 @@ pub mod requests;
 fn get_slack_url_for_method(method: &str) -> String {
     format!("https://slack.com/api/{}", method)
 }
+
+fn optional_struct_or_empty_array<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
+    where T: ::serde::Deserialize + Default,
+          D: ::serde::Deserializer
+{
+    use std::marker::PhantomData;
+    use serde::de;
+    use std::fmt;
+
+    #[derive(Default)]
+    struct StructOrEmptyArray<T>(PhantomData<T>);
```

Comment:
https://docs.serde.rs/serde/de/trait.DeserializeSeed.html mention of it here also